### PR TITLE
Edit own places

### PR DESCRIPTION
**Pull request by JanineGlaser**
you can now editing your own places
edit images will follow

### CODE REVIEW

File: components/Cards/DetailsCard.js [hunk 1]
```diff
@@ -170,52 +298,146 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
...
-  useEffect(() => {
-    // Fetch comments associated with the current place
-    fetchComments();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [_id]);
+  // useEffect(() => {
+  //   // Fetch comments associated with the current place
+  //   fetchComments();
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [_id]);
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:305]:
```suggestion
```

File: components/Cards/DetailsCard.js [hunk 2]
```diff
@@ -35,19 +47,135 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
...
+  async function handleEditPlaceSubmit() {
+    try {
+      // Get the edited place data
+      const editedPlaceText = getTextAreaValue("initialReview");
+      const editedDescriptionText = getTextAreaValue("description");
+      const editedPlaceName = getInputValue("placeName");
+      const editedRegion = getInputValue("region");
+      const editedActivities = editingActivities;
+
+      console.log(editedActivities);
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:114]:
```suggestion
```

File: components/Cards/DetailsCard.js [hunk 3]
```diff
@@ -35,19 +47,135 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
 
   const [comments, setComments] = useState([]);
   const [editingCommentId, setEditingCommentId] = useState(null);
+  const [editingPlaceId, setEditingPlaceId] = useState(false);
+  const [editingActivities, setEditingActivities] = useState([]);
+  const [editingImage, setEditingImage] = useState();
+  const [editingPublicId, setEditingPublicId] = useState();
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:53]: I know this is something you want to implement at a later point, so for this PR, you can remove those two states.
Or do you already use them someplace and I overlooked it?

File: components/Cards/DetailsCard.js [hunk 4]
```diff
@@ -35,19 +47,135 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
 
   const [comments, setComments] = useState([]);
   const [editingCommentId, setEditingCommentId] = useState(null);
+  const [editingPlaceId, setEditingPlaceId] = useState(false);
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:50]: the name of the state suggests to me that an ID / string is being stored here and not a boolean value.

Variables / States that store a boolean value are usually prefixed with `is`, `has`, `can`

File: pages/api/places/[id].js [hunk 5]
```diff
@@ -27,6 +27,23 @@ export default async function handler(request, response) {
...
   }
+  //----------------------------------------------------------------
+
+  if (request.method === "PUT") {
+    try {
+      const { initialReview, description, name, region, activities } =
+        request.body;
+      const updatedPlace = await Place.findByIdAndUpdate(
+        id,
+        { initialReview, description, name, region, activities },
```
Comment (NikRoe) [inline pages/api/places/[id].js:38]: mongoose should be clever enough to also allow this and only change the fields that need an update:

```suggestion
      const updatedPlaceData =
        request.body;
      const updatedPlace = await Place.findByIdAndUpdate(
        id,
        updatedPlaceData,
```

File: components/Cards/DetailsCard.js [hunk 6]
```diff
@@ -170,52 +298,146 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
...
-                </StyledSecondaryButton>
-              </StyledCardButtonWrapper>
+              <>
+                <StyledCardButtonWrapper>
+                  {editingPlaceId ? (
+                    <StyledSecondaryButton onClick={cancelEditPlace}>
+                      <FontAwesomeIcon icon={faTrashSolid} />
+                      &nbsp;cancel edit
+                    </StyledSecondaryButton>
+                  ) : null}
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:418]: You could also write all of these checks like this:

```suggestion
                  {editingPlaceId && (
                    <StyledSecondaryButton onClick={cancelEditPlace}>
                      <FontAwesomeIcon icon={faTrashSolid} />
                      &nbsp;cancel edit
                    </StyledSecondaryButton>
                  )}
```

This is also applicable for the following lines of code

File: components/Cards/DetailsCard.js [hunk 7]
```diff
@@ -35,19 +47,135 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
...
+  //----------------------------------------------------------------
+
+  // for the place
+  async function handleEditPlaceSubmit() {
+    try {
+      // Get the edited place data
+      const editedPlaceText = getTextAreaValue("initialReview");
+      const editedDescriptionText = getTextAreaValue("description");
+      const editedPlaceName = getInputValue("placeName");
+      const editedRegion = getInputValue("region");
```
Comment (NikRoe) [inline components/Cards/DetailsCard.js:111]: at a first glance this looks like more work than needed, however, I am assuming you first tried getting the data using `new FormData` and ran into issues that lead to write those functions in order to get the data from the form.

If that is not the case, I recommend you try getting the data of the form using FormData

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

nice and promising start! 🙌 

There are some lines that you can get rid of and I left some comments where you can rename some of your variables, reduce the amount of code or try out a different approach when getting the data.

Keep it up 🙌

File: components/Cards/DetailsCard.js (hunk 7 above)
Comment (JanineGlaser) [reply components/Cards/DetailsCard.js:111]: Hi Niklas, thank you for you review, we adjusted our code. The last suggestion caused  some issues, we will have a look at this later on. thank you :)

**Review by NikRoe — APPROVED** [review]
LGTM

**Pull request merged**